Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/event/model/Certificate.java

```diff
@@ -117,15 +138,45 @@ public void setVersion(int version) {
     }
 
     public String getSubjectKeyID() {
-        return subjectKeyID;
+        if (subjectKeyID == null) {
+            byte[] subjectKeyId = x509Cert.getExtensionValue(SUBJECT_KEY_ID);
```

Comment:
Why do we put all logic in this DTO bean? Should we refactor this into Util and both classess can use it?